### ISSUE TITLE
UI: sample table sorting/pagination interaction

### ISSUE THREAD

**Issue by cassimons** [populationgenomics/sample-metadata#363]
When a participant has multiple samples, the sample table will join these into a merged row - very handy to see that a single participant has multiple samples. However, this only seems to occur if pagination puts both samples on the same display page. Would it be practical to group/sort by the participant before pagination?

EG the 1st screen grab is `ag-hidden` with a [page size of 20](https://sample-metadata.populationgenomics.org.au/project/ag-hidden/1?size=20). Individual A0134002 is highlighted with a single sample. The second grab is of the same project and individual, but with [page size set to 100](https://sample-metadata.populationgenomics.org.au/project/ag-hidden/1?size=100), now both samples are displayed. 

![image](https://user-images.githubusercontent.com/21028507/205400647-eec6c095-f8f9-4a4d-a19d-3b8823e50417.png)

![image](https://user-images.githubusercontent.com/21028507/205400695-7d20325e-7969-454b-a549-7730a22b15a5.png)

**Comment by danielreti** [comment]
@illusional can probably unpack this a bit more but as I understand it, I don't think sorting would necessarily solve this in the current implementation given that it is a sample level breakdown. Eg what if the 20th sample on a page is from a participant with multiple samples. It would necessarily overflow to the next page and grouping would leave you with more than 20 samples on a page. Not saying that would be a terrible thing but it doesn't make sense currently. 

I agree though that it would be cleaner to have all of a participants samples grouped together but this may need more thought and may not be as trivial to solve fully as simply sorting beforehand. (Although it may solve some/most cases)

**Comment by illusional** [comment]
Yeah this isn't trivial, this page is formed by grabbing relevant samples, then filling in participant, family + sequence info. This view (by design) does NOT show sequences, participants families if there is no sample to grab them - and looking ahead is not trivial.

This query does group participants regardless of where they appear within the query. Although I can see the ideal, I don't think it's practical in the short or medium term.

**Comment by cassimons** [comment]
No drama it is not as easy fix, it is absolutely not essential. Makes me hungry for the participant-based view though 🤤.
